optaweb-vehicle-routing-backend/src/main/java/org/optaweb/vehiclerouting/service/demo/RoutingProblemConfig.java

```diff
@@ -62,38 +60,19 @@
 
     @Produces
     RoutingProblemList routingProblems() {
-        ArrayList<RoutingProblem> problems = new ArrayList<>();
-        problems.add(dataSetMarshaller.unmarshal(belgiumReader()));
-        problems.addAll(localDataSets());
-        return new RoutingProblemList(problems);
+        Stream<RoutingProblem> allProblems = Stream.concat(classPathProblems(), dataSetDirProblems());
```

Comment:
This is the important change. Instead of building a new array list, simply join two streams.